x-pack/plugins/security_solution/public/management/pages/trusted_apps/view/trusted_apps_list.test.tsx

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { combineReducers, createStore } from 'redux';
+import { render } from '@testing-library/react';
+import React from 'react';
+import { Provider } from 'react-redux';
+
+import {
+  MANAGEMENT_STORE_GLOBAL_NAMESPACE,
+  MANAGEMENT_STORE_TRUSTED_APPS_NAMESPACE,
+} from '../../../common/constants';
+import { trustedAppsPageReducer } from '../store/reducer';
+import { TrustedAppsList } from './trusted_apps_list';
+import {
+  createListFailedResourceState,
+  createListLoadedResourceState,
+  createListLoadingResourceState,
+  createTrustedAppsListResourceStateChangedAction,
+  createUserChangedUrlAction,
+} from '../test_utils';
+
+jest.mock('@elastic/eui/lib/services/accessibility/html_id_generator', () => ({
+  htmlIdGenerator: () => () => 'mockId',
+}));
+
+const createStoreSetup = () => {
```

Comment:
I saw that utility and explored it quite a lot, you can notice quite some of my stuff is inspired by it, from the other side I was trying to explore if I can keep things simple for the start. I think with time we will just converge to using the full utility. What would be nice is to split that utility so that parts of it could be reused.

My observation was that if majority of the logic is done in Redux, the testing of such components do not even require middleware and such and only needs state and reducer to reproduce any possible state of the component.